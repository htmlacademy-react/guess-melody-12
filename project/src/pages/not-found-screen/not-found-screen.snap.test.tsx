import {render} from '@testing-library/react';
import {HelmetProvider} from 'react-helmet-async';
import {createMemoryHistory} from 'history';
import NotFoundScreen from './not-found-screen';
import HistoryRouter from '../../components/history-route/history-route';

describe('Component: NotFoundScreen', () => {
  it('should render correctly', () => {
    const history = createMemoryHistory();

    const {container} = render(
      <HistoryRouter history={history}>
        <HelmetProvider>
          <NotFoundScreen />
        </HelmetProvider>
      </HistoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
