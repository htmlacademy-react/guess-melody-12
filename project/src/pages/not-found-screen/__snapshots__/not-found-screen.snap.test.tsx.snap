// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: NotFoundScreen should render correctly 1`] = `
<div>
  <section
    class="game"
  >
    <header
      class="game__header"
    >
      <a
        class="game__back"
        href="/"
      >
        <span
          class="visually-hidden"
        >
          Сыграть ещё раз
        </span>
        <img
          alt="Угадай мелодию"
          class="game__logo"
          src="img/melody-logo-ginger.png"
        />
      </a>
      <svg
        class="timer"
        viewBox="0 0 780 780"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          class="timer__line"
          cx="390"
          cy="390"
          r="370"
          style="filter: url(#blur); transform: rotate(-90deg) scaleY(-1); transform-origin: center;"
        />
      </svg>
    </header>
    <section
      class="game__screen"
    >
      <h1>
        404. Page not found
      </h1>
      <a
        href="/"
      >
        Вернуться на главную
      </a>
    </section>
  </section>
</div>
`;
